### DISCUSSION TITLE
Absolute values for magnetic moments

### DISCUSSION THREAD

**Discussion by sebastianecheverrir** [CederGroupHub/chgnet#98] [Q&A]
In lines 118-124 of the file [dataset.py ](https://github.com/CederGroupHub/chgnet/blob/main/chgnet/data/dataset.py) the absolute values of the magnetic moments is taken
```
if self.magmoms is not None:
    mag = self.magmoms[graph_id]
    # use absolute value for magnetic moments
    if mag is None:
        targets["m"] = None
    else:
        targets["m"] = torch.abs(torch.tensor(mag, dtype=datatype))
```
It is also mentioned in the [CHG Nature article ](https://www.nature.com/articles/s42256-023-00716-3) that "The absolute values of DFT magmoms are used for training"

Is there a physical or numerical explanation for this?

**Comment by bowen-bd** [comment]
The signs of MAGMOMs come from the direction of initialization.
That is to say, multiply all the MAGMOMs by -1 won't change anything about this system.

CHGNet is not able to predict a plus or minus value.
To avoid this ambiguity in training, we use the absolute value of MAGMOM.

**Reply by naveenmohandas** [reply]
Wouldn't it wrongly map energy for a  paramagnetic structure with two atoms, having magmom [ -m m ], as it takes the absolute value [ m m] for training?
Or is CHGnet only suited for ferromagnetic structures?

**Reply by bowen-bd** [reply]
@naveenmohandas 
The current implementation does not support predicting the energy associated with different spins.
CHGNet is trained on the ground state spin and its associated energy from Materials Project.

if multiple spin states and associated energies exists for same atomic configurations, we only kept the energy ground state in the MPtrj dataset.
More data paring details are discussed in #97
